table/column.go

```diff
@@ -182,7 +182,9 @@ func CastValue(ctx sessionctx.Context, val types.Datum, col *model.ColumnInfo) (
 	} else {
 		err = sc.HandleTruncate(err)
 	}
-	if err != nil {
+	if forceIgnoreTruncate {
```

Comment:
Better to check specific error types and return nil for these errors only

Reply:
Or put the error check logic where `CastFunc` is used for the virtual column, and this will not need to change the `CastFunc` signature and parameters

Reply:
Yes, it is better to check the error type.